### PR TITLE
Update c/storage to latest release

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
This bump fixes a runtime panic when trying to access the layers
lockfile and its parent directory does not exist.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Fixed runtime panic for layers lockfile if its parent directory does not exist.
Added support for repositories in auth.json
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

**Comment by saschagrunert** [comment]
@haircommander PTAL

**Comment by haircommander** [comment]
LGTM

**Comment by saschagrunert** [comment]
Had to rebase

**Comment by fidencio** [comment]
@saschagrunert, the failure on kata-containers here seems to be legit and related to this PR:
```
18:13:52 ./bin/crio -d "" --config=""  config > crio.conf
18:13:52 error loading server config: lstat /run/containers/storage: permission denied
18:13:52 make: *** [Makefile:188: crio.conf] Error 1
18:13:52 [install_crio.sh:126] ERROR: make BUILDTAGS=seccomp selinux exclude_graphdriver_btrfs exclude_graphdriver_devicemapper libdm_no_deferred_remove
18:13:52 [setup.sh:116] ERROR: CRI-O not installe
```

Any idea why this PR could be causing that?  I guess this is easy to reproduce locally, without kata, with a CentOS 8 VM.

File: go.mod [hunk 1]
```diff
@@ -13,13 +13,13 @@ require (
...
+	github.com/containers/buildah v1.22.0
+	github.com/containers/common v0.43.0
 	github.com/containers/conmon v2.0.20+incompatible
-	github.com/containers/image/v5 v5.13.2
+	github.com/containers/image/v5 v5.15.0
 	github.com/containers/ocicrypt v1.1.2
-	github.com/containers/podman/v3 v3.2.3
-	github.com/containers/storage v1.32.3
+	github.com/containers/podman/v3 v3.3.0-rc1
+	github.com/containers/storage v1.34.0
```
Comment (TomSweeneyRedHat) [inline go.mod:22]: now up to c1.34.1, you might want to get that, @giuseppe found an issue with fuse-overlayfs that he squashed.

**Comment by TomSweeneyRedHat** [comment]
And it looks like #5210 is bumping to v1.34.1 too.

File: go.mod (hunk 1 above)
Comment (saschagrunert) [reply go.mod]: Updated as suggested :+1:

**Comment by saschagrunert** [comment]
@haircommander PTAL

**Comment by saschagrunert** [comment]
Had to rebase because of a merge conflict.

**Pull request merged**